Lombiq.HelpfulLibraries/Libraries/Database/ManualConnectingIndexService.cs

```diff
@@ -0,0 +1,100 @@
+using Dapper;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using OrchardCore.Data;
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using YesSql;
+using YesSql.Indexes;
+
+namespace Lombiq.HelpfulLibraries.Libraries.Database
+{
+    public class ManualConnectingIndexService<T> : IManualConnectingIndexService<T>
+        where T : MapIndex
+    {
+        private readonly Type _type;
+        private readonly Dictionary<string, PropertyInfo> _properties;
+        private readonly IDbConnectionAccessor _dbAccessor;
+        private readonly ILogger _logger;
+
+        private string _documentIdKey;
+        private string _keys;
+        private string _columns;
+        private string _tablePrefix;
+
+        public ManualConnectingIndexService(
+            IDbConnectionAccessor dbAccessor,
+            ILogger<ManualConnectingIndexService<T>> logger)
+        {
+            _type = typeof(T);
+            _properties = _type
+                .GetProperties()
+                .Where(property => property.Name != nameof(MapIndex.Id))
+                .ToDictionary(x => x.Name);
+
+            _dbAccessor = dbAccessor;
+            _logger = logger;
+        }
+
+        public Task AddAsync(T item, ISession session, int? setDocumentId = null) =>
+            RunTransactionAsync(session, async (connection, transaction, dialect, name) =>
+            {
+                _documentIdKey ??= dialect.QuoteForColumnName("DocumentId");
+                _keys ??= string.Join(", ", _properties.Keys.Select(key => "@" + key));
+                _columns ??= string.Join(", ", _properties.Keys.Select(dialect.QuoteForColumnName));
+
+                var documentId = setDocumentId ?? (item as IIndex).GetAddedDocuments().Single().Id;
+                var sql = $"INSERT INTO {name} ({_documentIdKey}, {_columns}) VALUES ({dialect.GetSqlValue(documentId)}, {_keys});";

```

Comment:
A `_keys` nem SQL injection lehetőség? A többi kódból jön, de ez adat és bármi lehet.

Reply:
A `_keys` is csak kódból jön, reflectionnel olvassa ki a `T` property neveit. Viszont nem tudom miért nem a ctor-ban van az értékadás. Megpróbálom áttenni.